### PR TITLE
Compress shell script and remove dependency on ramdisks.

### PR DESCRIPTION
**Pull request by MichaelRawson**
Should be done entirely in memory (pipes specifically) now and all stages in parallel. I can't test this properly as I don't have any hardware, but believe it should work. Tests please!

If this works, moving it inside the main server should be easy.

### CODE REVIEW

File: wspr_rx.sh [hunk 1]
```diff
@@ -9,24 +9,20 @@ then
...
-sudo arecord -f S16_LE -r 48000 --duration=114 -q -t wav -D hw:0,0 -c 2 /mnt/ramdisk/raw.wav
-echo "Recording complete"
-sox /mnt/ramdisk/raw.wav -c 1 -r 12000 output.wav
-echo "Decimation complete"
-~/wspr_rx/wsprd -wf "$3" test.wav >/dev/null 2>&1
-sed -i 's/tput   //g' ALL_WSPR.TXT
-sed -i "s/^....../$start_time/g" ALL_WSPR.TXT
+
+echo "Recording..."
+arecord -f S16_LE -r 48000 --duration=114 -q -t wav -D hw:0,0 -c 2 | sox -t wav -c 1 -r 12000 - - | wsprd -wf "$3" /dev/stdin >/dev/null 2>&1
```
Comment (MichaelRawson) [inline wspr_rx.sh]: Can `wsprd` change the output file? If so we can have it write to `/dev/stdout` straight away and pipe it to `sed`. Even more pipes!

File: wspr_rx.sh [hunk 2]
```diff
@@ -9,24 +9,20 @@ then
...
+echo "...done."
+
 echo ""
 echo ""
-cat ALL_WSPR.TXT
+sed -i "s/tput   //g;s/^....../$start_time/g" < ALL_WSPR.TXT
```
Comment (MichaelRawson) [inline wspr_rx.sh]: Derp this should be without the `-i`.

File: wspr_rx.sh [hunk 3]
```diff
@@ -9,24 +9,20 @@ then
...
 echo ""
-#curl -F allmept=@ALL_WSPR.TXT -F call="$1" -F grid="$2" wsprnet.org/meptspots.php >/dev/null 2>&1
-echo "Upload complete"
-rm ALL_WSPR.TXT
-rm raw.wav
-rm output.wav
-
 
+echo "Uploading..."
+curl -F allmept=@ALL_WSPR.TXT -F call="$1" -F grid="$2" wsprnet.org/meptspots.php >/dev/null 2>&1
```
Comment (MichaelRawson) [inline wspr_rx.sh]: This will likely be done in software if we can.

File: wspr_rx.sh (hunk 1 above)
Comment (M0WUT) [reply wspr_rx.sh]: Not sure, the c source code that it is running is here if you are interested / think that may be a better way to do it. It's a bit grim as that guy loves his variable names to be... concise, shall we say: https://github.com/k9an/old_wsprcan Line of interest is 422 in wsprd.c

Comment (M0WUT) [reply wspr_rx.sh]: By not sure, I mean. No, it's hard coded but you may be able to edit the source to allow it, not sure if fopen works with stdout

**Review by M0WUT — CHANGES_REQUESTED** [review]
Code fails instantly, sox is not waiting for arecord to complete and gets called instantly. Error message: "pi@WSPRpi:~/wspr_rx $ ./wspr_rx.sh m0wut io93fp 14.0956
./wspr_rx.sh: 4: ./wspr_rx.sh: [[: not found
WSPR Decoder by M0WUT: Callsign = m0wut, Locator = io93fp, Frequency = 14.0956
Recording...
sox FAIL formats: can't determine type of `-'
...done.


./wspr_rx.sh: 21: ./wspr_rx.sh: cannot open ALL_WSPR.TXT: No such file


Uploading...
...done.
rm: cannot remove ‘ALL_WSPR.TXT’: No such file or directory"

P.s. This can be run on a standard linux machine, nothing hardware specific, at least my original script runs on my Linux VM without any changes. (other than wsprd not found)

File: wspr_rx.sh (hunk 1 above)
Comment (MichaelRawson) [reply wspr_rx.sh]: Ah, don't worry then. `fopen` will work with any file, but if it's hard-coded probably easier just to open the file.

**Comment by MichaelRawson** [comment]
So I got this to work by (1) testing it properly (sorry...) and (2) bodging the WSPR decoder a bit - see the new repository. Should now be callable from a python script with a few further tweaks without any problems - testing appreciated as (while it runs now) I obviously can't get any output from it.

**Review by M0WUT — APPROVED** [review]

**Pull request merged**